Sources/WhisperKit/Core/Models.swift

```diff
@@ -836,52 +836,73 @@ public class TextDecoderCachePrefillOutput: MLFeatureProvider {
+    public let noTimestampsToken: Int
+    public let specialTokenBegin: Int
+    public let startOfTranscriptToken: Int
+    public let timeTokenBegin: Int
+    public let transcribeToken: Int
+    public let translateToken: Int
+    public let whitespaceToken: Int
 
-    /// Default values for each token, using base vocab
-    internal static var defaultWhitespaceToken: Int { 220 }
-    internal static var defaultSpecialTokenBegin: Int { 50257 }
-    internal static var defaultEndToken: Int { 50257 }
-    internal static var defaultStartOfTranscriptToken: Int { 50258 }
-    internal static var defaultEnglishToken: Int { 50259 }
-    internal static var defaultTranscribeToken: Int { 50359 }
-    internal static var defaultTranslateToken: Int { 50358 }
-    internal static var defaultNoSpeechToken: Int { 50362 }
-    internal static var defaultNoTimestampsToken: Int { 50363 }
-    internal static var defaultTimeTokenBegin: Int { 50364 }
-
-    /// Tokenizes the given text into individual words and associated tokens
-    /// - Parameter tokenIds: Array of tokens to split
-    /// - Returns: Tuple containing and array of the split words and and all tokens for each word
-    func splitToWordTokens(tokenIds: [Int]) -> (words: [String], wordTokens: [[Int]]) {
-        let decodedWords = decode(tokens: tokenIds.filter { $0 < specialTokenBegin })
-
-        // Detect language of input text
-        let recognizer = NLLanguageRecognizer()
-        recognizer.processString(decodedWords)
-        let languageCode = recognizer.dominantLanguage?.rawValue
-
-        if ["zh", "ja", "th", "lo", "my", "yue"].contains(languageCode) {
-            return splitTokensOnUnicode(tokens: tokenIds)
-        } else {
-            return splitTokensOnSpaces(tokens: tokenIds)
-        }
+    public init(
+        endToken: Int,
+        englishToken: Int,
+        noSpeechToken: Int,
+        noTimestampsToken: Int,
+        specialTokenBegin: Int,
+        startOfTranscriptToken: Int,
+        timeTokenBegin: Int,
+        transcribeToken: Int,
+        translateToken: Int,
+        whitespaceToken: Int
+    ) {
+        self.endToken = endToken
+        self.englishToken = englishToken
+        self.noSpeechToken = noSpeechToken
+        self.noTimestampsToken = noTimestampsToken
+        self.specialTokenBegin = specialTokenBegin
+        self.startOfTranscriptToken = startOfTranscriptToken
+        self.timeTokenBegin = timeTokenBegin
+        self.transcribeToken = transcribeToken
+        self.translateToken = translateToken
+        self.whitespaceToken = whitespaceToken
     }
+}
 
-    func splitTokensOnUnicode(tokens: [Int]) -> (words: [String], wordTokens: [[Int]]) {
-        let decodedFull = decode(tokens: tokens)
+public protocol WhisperTokenizer: Tokenizer {
+    var specialTokens: SpecialTokens { get }
+    
+    func splitToWordTokens(tokenIds: [Int]) -> (words: [String], wordTokens: [[Int]])
+}
+
+struct WhisperTokenizerWrapper: WhisperTokenizer {
```

Comment:
Is there anything that could be done in `swift-transformers` to make this easier to extend?

Reply:
not sure if there is anything that can be done, any thoughts @ZachNagengast?

Reply:
we just wanted to have an efficient store for special tokens which are commonly used throughout the codebase. Extending `swift-transformers` `Tokenizer` protocol has its limitations -- only computed properties are allowed so I decided to go this way

Reply:
Yes, I wonder if we should have used classes in some places :)

Reply:
Special tokens would be nice to have directly from the config. We have one issue with the whisper tokenizer where they changed the no speech token to be <|nocaptions|> on some models and <|nospeech|> on others, and the way we're initializing right now is actually encoding this string, which is not resilient to that change.